### PR TITLE
jwt: fix claim to header example doc

### PR DESCRIPTION
**Pull request by cainelli**
Commit Message: jwt: fix claim to header example doc
Additional Description: The example is not [correct](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/jwt_authn/v3/config.proto#extensions-filters-http-jwt-authn-v3-jwtclaimtoheader), got bitten by copy/paste.
Risk Level: Low

### CODE REVIEW

**Comment by phlax** [comment]
@cainelli there is a way to make sure the examples _are_ correct - but for api examples its non-trivial - essentially moving the example into a full bootstrap config and using `literalinclude` to add to the docs. In that case the configs get tested in ci

**Comment by cainelli** [comment]
aha! You mean like this one?
 https://github.com/envoyproxy/envoy/blob/34bdb945cb8a77630363659c24baf21e9aaf891f/docs/root/configuration/http/cluster_specifier/lua.rst?plain=1#L27-L32

I can adjust to this approach.

**Comment by phlax** [comment]
yep - exactly like that - the only difference here is that its in the proto file - so you would have to figure a good place to put the config and then figure out the path to it

if you are up for doing that it would be really appreciated - it means we can have confidence that the example works now, and will stay working

File: api/envoy/extensions/filters/http/jwt_authn/v3/config.proto [hunk 1]
```diff
@@ -355,11 +355,9 @@ message JwtProvider {
...
   // The claim must be of type; string, int, double, bool. Array type claims are not supported
   //
-  // .. code-block:: yaml
-  //
-  //   claim_to_headers:
-  //     - name: x-jwt-claim-nested-claim
-  //       claim: claim.nested.key
+  // .. literalinclude:: /_configs/repo/jwt_authn.yaml
+  //    :language: yaml
+  //    :lines: 44-48
```
Comment (cainelli) [inline api/envoy/extensions/filters/http/jwt_authn/v3/config.proto:360]: image

**Comment by cainelli** [comment]
@phlax  if this sounds reasonable let me know if I should change the other examples as well.

**Review by abeyad — DISMISSED** [review]
/lgtm api

once @phlax approves, it's good to go

File: api/envoy/extensions/filters/http/jwt_authn/v3/config.proto (hunk 1 above)
Comment (phlax) [inline api/envoy/extensions/filters/http/jwt_authn/v3/config.proto:360]:
```suggestion
  //    :lines: 44-48
  //    :linenos
  //    :lineno-start: 44
  //    :caption: :download:`jwt_authn.yaml </_configs/repo/jwt_authn.yaml>`
```

**Review by phlax — COMMENTED** [review]
great! thanks @cainelli 

i think lets add line numbers and a download option as that is really useful for end users trying to figure out how to use it

File: api/envoy/extensions/filters/http/jwt_authn/v3/config.proto [hunk 2]
```diff
@@ -355,11 +355,12 @@ message JwtProvider {
...
+  //    :linenos
```
Comment (phlax) [inline api/envoy/extensions/filters/http/jwt_authn/v3/config.proto:361]:
```suggestion
  //    :linenos:
```

apologies - seems it needs an ending colon

also there is a format issue which should be fixed with:

```diff

diff --git a/tools/spelling/check_spelling_pedantic.py b/tools/spelling/check_spelling_pedantic.py
index 2ea0a70..082122d 100755
--- a/tools/spelling/check_spelling_pedantic.py
+++ b/tools/spelling/check_spelling_pedantic.py
@@ -690,7 +690,8 @@ def extract_comments(lines):
                     break
 
                 n += 1
-        elif text.strip().startswith(RST_CODE_BLOCK) or text.strip().startswith(RST_LITERAL_INCLUDE):
+        elif text.strip().startswith(RST_CODE_BLOCK) or text.strip().startswith(
+                RST_LITERAL_INCLUDE):
             # Start of a code block.
             indent = len(INDENT.search(text).group(1))
             last_line = comments[n].line
```

**Review by phlax — DISMISSED** [review]
lgtm, thanks @cainelli - really appreciated

regarding updating other configs that would be amazing, but lets deal with them in a follow up

**Comment by cainelli** [comment]
> regarding updating other configs that would be amazing, but lets deal with them in a follow up

Sounds good to me. I've pushed a [fix](https://github.com/envoyproxy/envoy/pull/35532/commits/9f1cd0615ca71469be9288f4e5c25dbed10da1f9) for the CI issue.

**Review by phlax — APPROVED** [review]
lgtm, thanks again @cainelli

**Pull request merged**